hyperspy/tests/drawing/test_mpl_testing_setup.py

```diff
@@ -16,14 +16,40 @@
 # along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
 
 import matplotlib
-import pytest
 from distutils.version import LooseVersion
+import pytest
+
+from hyperspy.misc.test_utils import check_running_tests_in_CI
 
 
-def test_mlp_agg_for_testing():
-    assert matplotlib.get_backend() == 'agg'
+def test_mlp_agg_for_CI_testing():
+    if check_running_tests_in_CI():
+        assert matplotlib.get_backend() == 'agg'
 
 
 def test_mpl_version():
     # for simplicity, only matplotlib 2.x is supported for testing
     assert LooseVersion(matplotlib.__version__) >= LooseVersion('2.0.0')
+
+
+# Skip if mpl plugin is not called, because it will not failed (no image comparison)
+@pytest.mark.skipif(not pytest.config.getvalue("mpl"),
+                    reason="'mpl' plugin not call.")
+@pytest.mark.xfail(reason="Check if plotting tests are working: if this failed,"
+                   "it means that the image comparison of the plotting test are"
+                   " not working.",
```

Comment:
Here the `reason` and commented explanations do not match. I assume it's just the wording of the reason that could be improved?

Also, `skipif` reason should have `"...call"` -> `"...called"`